### PR TITLE
feat(typedoc-appium-plugin): implement cross-referencing of methods

### PR DESCRIPTION
**Pull request by boneskull**
This is getting pretty close, but:

- Commands from a driver will show even w/o a `newMethodMap` or `execMethodMap`; it is able to understand that it's overriding `BaseDriver`
- It's also able to understand the associated routes
- And it's able to understand the routes even if `BaseDriver` does not implement it

~~Still working on the parameters display, but we have the name override working at least.~~

**Update:** Parameters are working fine.

Added some docstrings in `fake-driver`, `base-driver` for testing

### CODE REVIEW

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17971** <a href="https://app.graphite.dev/github/pr/appium/appium/17971" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium/17971?utm_source=stack-comment).

File: packages/typedoc-plugin-appium/lib/converter/comment.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+/**
+ * Utilities to derive comments from various sources
+ * @module
+ */
+
+import _ from 'lodash';
+import {Comment, CommentDisplayPart} from 'typedoc';
+import {AsyncMethodDeclarationReflection, KnownMethods} from './types';
+
+export const stats = _.fill(new Array(5), 0);
```
Comment (boneskull) [inline packages/typedoc-plugin-appium/lib/converter/comment.ts:10]: this is for debugging to let me know which "comment finders" are actually being called, so I can remove the ones that don't get called

**Review by jlipps — COMMENTED** [review]
just saving my review state

**Comment by boneskull** [comment]
@jlipps OK, this is looking pretty good now.  This is what we get from `FakeDriver`:

image

Above we can see that:

1. The comment comes from the `FakeDriver#createSession()` method; it is not in the method map.
2. The parameter names and whether they are required or optional come from the _builtin method map_ in `routes.js`.  
3. The capabilities types come from `FakeDriver` 

Here are some from `FakeDriver`'s method map:

image

1. The `GET` method of `/session/:sessionId/fakedriver` has no description in the new method map _or_ the method itself.
2. The `POST` method of `/session/:sessionId/fakedriver` has a description from the method itself, which takes precedence over the description from the method map (which exists)

And some from the execute method map:

image

1. The description of `fake: addition` and `fake: setThing` come from the class methods associated with the execute method
2. The description of `fake: getThing` comes from the execute method map itself, since the class method has no description

**Comment by boneskull** [comment]
And there's some output when run with verbose log level, so you can see what's happening:

```
[appium:converter] Converting module @appium/fake-driver
[appium:converter] (FakeDriver) Converting 40 potential methods
[appium:converter] (FakeDriver) Added GET route /session/:sessionId/fakedriver to method getFakeThing for command "getFakeThing"
[appium:converter] (FakeDriver) Added POST route /session/:sessionId/fakedriver to method setFakeThing for command "setFakeThing"
[appium:converter] (FakeDriver) Added GET route /session/:sessionId/fakedriverargs to method getFakeDriverArgs for command "getFakeDriverArgs"
[appium:converter] (FakeDriver) Added /session/:sessionId/execute route to method fakeAddition for command "fakeAddition" from script "fake: addition"
[appium:converter] (FakeDriver) Added /session/:sessionId/execute route to method getFakeThing for command "getFakeThing" from script "fake: getThing"
[appium:converter] (FakeDriver) Added /session/:sessionId/execute route to method setFakeThing for command "setFakeThing" from script "fake: setThing"
[appium:converter] (FakeDriver) Added POST route to method createSession for command "createSession"
[appium:converter] (FakeDriver) Added POST route to method reset for command "reset"
[appium:converter] (FakeDriver) Added GET route to method deleteSession for command "deleteSession"
[appium:converter] (FakeDriver) Added DELETE route to method deleteSession for command "deleteSession"
[appium:converter] (FakeDriver) Added GET route to method getStatus for command "getStatus"
[appium:converter] (FakeDriver) Added GET route to method timeouts for command "timeouts"
[appium:converter] (FakeDriver) Added POST route to method timeouts for command "timeouts"
[appium:converter] (FakeDriver) Added POST route to method implicitWait for command "implicitWait"
[appium:converter] (FakeDriver) Added GET route to method getTimeouts for command "getTimeouts"
[appium:converter] (FakeDriver) Added POST route to method getTimeouts for command "getTimeouts"
[appium:converter] (FakeDriver) Added POST route to method logCustomEvent for command "logCustomEvent"
[appium:converter] (FakeDriver) Added POST route to method getLogEvents for command "getLogEvents"
[appium:converter] (FakeDriver) Added POST route to method findElement for command "findElement"
[appium:converter] (FakeDriver) Added POST route to method findElements for command "findElements"
[appium:converter] (FakeDriver) Added POST route to method findElementFromElement for command "findElementFromElement"
[appium:converter] (FakeDriver) Added POST route to method findElementsFromElement for command "findElementsFromElement"
[appium:converter] (FakeDriver) Added GET route to method getPageSource for command "getPageSource"
[appium:converter] (FakeDriver) Added GET route to method getLogTypes for command "getLogTypes"
[appium:converter] (FakeDriver) Added GET route to method getLogTypes for command "getLogTypes"
[appium:converter] (FakeDriver) Added POST route to method getLog for command "getLog"
[appium:converter] (FakeDriver) Added POST route to method getLog for command "getLog"
[appium:converter] (FakeDriver) Added POST route to method getSettings for command "getSettings"
[appium:converter] (FakeDriver) Added GET route to method getSettings for command "getSettings"
[appium:converter] (FakeDriver) Added GET route to method getSessions for command "getSessions"
[appium:converter] (FakeDriver) Added GET route to method getSession for command "getSession"
[appium:converter] (FakeDriver) Added DELETE route to method getSession for command "getSession"
[appium:converter] (FakeDriver) The following async methods were not found in any method map: updateServer, proxyCommand, executeCommand, startUnexpectedShutdown, startNewCommandTimeout, clearNewCommandTimeout, implicitWaitForCondition, implicitWaitW3C, implicitWaitMJSONWP, pageLoadTimeoutW3C, pageLoadTimeoutMJSONWP, scriptTimeoutW3C, scriptTimeoutMJSONWP, newCommandTimeout, findElOrEls, findElOrElsWithProcessing, executeMethod
[appium:converter] (FakeDriver) Done; found 21 routes and 3 execute methods
```

**Comment by boneskull** [comment]
("routes", for statistical purposes above, have a unique name, but each may have 1-3 associated HTTP methods)

**Review by jlipps — APPROVED** [review]
LGTM!

**Pull request merged**